x-pack/plugin/src/test/resources/rest-api-spec/api/slm.delete_lifecycle.json

```diff
@@ -1,13 +1,25 @@
 {
   "slm.delete_lifecycle":{
     "documentation":{
-      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/slm-api.html"
+      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/slm-api-delete.html"
     },
     "stability":"stable",
     "url":{
       "paths":[
         {
           "path":"/_slm/policy/{policy_id}",
+          "methods":[
+            "DELETE"
+          ],
+          "parts":{
+            "policy_id":{
+              "type":"string",
+              "description":"The id of the snapshot lifecycle policy to remove"
+            }
+          }
+        },
+        {
+          "path":"/_slm/policy",
```

Comment:
SLM delete policy does not support a path without the `{policy_id}` parameter, it's not allowed at all

Reply:
Thanks @dakrone! I was under the impression that a path without the `policy_id` was allowed, because in the original implementation, before the new spec format, the `policy_id` was not marked as `required`: https://github.com/elastic/elasticsearch/blob/a2e0db7783055300de77f3a0e525a21ac2bd7530/x-pack/plugin/src/test/resources/rest-api-spec/api/slm.delete_lifecycle.json

I'll update the spec.